File: CSharp/SpaceVIL/SpaceVIL/Items/LinesContainer.cs
using System.Drawing;
using System.Collections.Generic;
using SpaceVIL.Core;

namespace SpaceVIL
{
    /// <summary>
    /// LinesContainer is class for rendering lines in graph.
    /// </summary>
    public class LinesContainer : Primitive, ILines
    {
        static int count = 0;

        /// <summary>
        /// Default LinesContainer constructor.
        /// </summary>
        public LinesContainer() : base(name: "LinesContainer_" + count)
        {
            SetBackground(Color.Transparent);
            count++;
        }
        private float _lineThickness = 1.0f;
        /// <summary>
        /// Setting thickness of lines.
        /// </summary>
        /// <param name="thickness">Line thickness.</param>
        public void SetLineThickness(float thickness)
        {
            _lineThickness = thickness;
        }
        /// <summary>
        /// Getting lines thickness.
        /// </summary>
        /// <returns>Lines thickness.</returns>
        public float GetLineThickness()
        {
            return _lineThickness;
        }
        private Color _lineColor = Color.Blue;
        /// <summary>
        /// Setting lines color.
        /// <pata/> Default: Blue.
        /// </summary>
        /// <param name="color">Line color.</param>
        public void SetLineColor(Color color)
        {
            _lineColor = color;
        }
        /// <summary>
        /// Getting lines color.
        /// </summary>
        /// <returns>Lines color.</returns>
        public Color GetLineColor()
        {
            return _lineColor;
        }
        /// <summary>
        /// Setting points coordinates.
        /// </summary>
        /// <param name="coord">Points list as List of float[2] array.</param>
        public void SetPoints(List<float[]> coord)
        {
            List<float[]> tmp = new List<float[]>(coord);
            SetTriangles(tmp);
        }
        /// <summary>
        /// Getting points coordinates.
        /// </summary>
        /// <returns>Points list as List of float[2] array.</returns>
        public List<float[]> GetPoints()
        {
            if (GetTriangles() == null || GetTriangles().Count < 2)
                return null;
            return GetTriangles();
        }
        /// <summary>
        /// Overridden method for stretching the points position relative 
        /// to the current size of the item. 
        /// Use in conjunction with GetTriangles() and SetTriangles() methods.
        /// <para/> Notice: This method is mainly for overriding only. SpaceVIL calls 
        /// this method if necessary and no need to call it manually.
        /// </summary>
        public override void MakeShape()
        {
            if (GetTriangles() != null)
                SetTriangles(UpdateShape());
        }
    }
}

File: Java/SpaceVIL/spacevil/src/main/java/com/spvessel/spacevil/LinesContainer.java
package com.spvessel.spacevil;

import com.spvessel.spacevil.Core.InterfaceLines;

import java.util.List;
import java.util.LinkedList;
import java.awt.Color;

/**
 * LinesContainer is class for rendering lines in graph.
 */
public class LinesContainer extends Primitive implements InterfaceLines {
    private static int count = 0;

    /**
     * Default LinesContainer constructor.
     */
    public LinesContainer() {
        setItemName("LinesContainer_" + count);
        setBackground(0, 0, 0, 0);
        count++;
    }

    /**
     * Setting points coordinates.
     * 
     * @param coord Points list as List of float[2] array.
     */
    public void setPoints(List<float[]> coord) {
        List<float[]> tmp = new LinkedList<>(coord);
        setTriangles(tmp);
    }

    /**
     * Getting points coordinates.
     * 
     * @return Points list as List of float[2] array.
     */
    @Override
    public List<float[]> getPoints() {
        if (getTriangles() == null || getTriangles().size() < 2)
            return null;
        return getTriangles();
    }

    /**
     * Overridden method for stretching the points position relative to the current
     * size of the item. Use in conjunction with getTriangles() and setTriangles()
     * methods.
     * <p>
     * Notice: This method is mainly for overriding only. SpaceVIL calls this method
     * if necessary and no need to call it manually.
     */
    @Override
    public void makeShape() {
        if (getTriangles() != null)
            setTriangles(updateShape());
    }

    private float _lineThickness = 1.0f;

    /**
     * Setting thickness of lines.
     * 
     * @param thickness Line thickness.
     */
    public void setLineThickness(float thickness) {
        _lineThickness = thickness;
    }

    /**
     * Getting lines thickness.
     * 
     * @return Lines thickness.
     */
    public float getLineThickness() {
        return _lineThickness;
    }

    private Color _lineColor = new Color(0, 0, 255);

    /**
     * Setting lines color.
     * <p>
     * Default: Blue.
     * 
     * @param color Line color.
     */
    public void setLineColor(Color color) {
        _lineColor = color;
    }

    /**
     * Getting lines color.
     * 
     * @return Lines color.
     */
    public Color getLineColor() {
        return _lineColor;
    }
}